src/Elastic.Apm/Config/AbstractConfigurationReader.cs

```diff
@@ -669,6 +670,79 @@ private bool ParseBoolOption(ConfigurationKeyValue kv, bool defaultValue, string
 			return parsedValue;
 		}
 
+		protected IReadOnlyDictionary<string, string> ParseGlobalLabels(ConfigurationKeyValue kv) => ParseKeyValuePairs(kv, "GlobalLabels");
+
+		private IReadOnlyDictionary<string, string> ParseKeyValuePairs(ConfigurationKeyValue kv, string dbgOptionName)
+		{
+			//
+			// Value format:
+			// 					key=value[,key=value[,...]]
+			//
+
+			if (kv?.Value == null)
+			{
+				_logger?.Debug()?.Log(dbgOptionName + " configuration option doesn't have a valid value - using default (empty map)");
+				return new Dictionary<string, string>();
+			}
+
+			// Treat empty string as zero key-value pairs
+			var pairs = kv.Value.IsEmpty() ? Array.Empty<string>() : kv.Value.Split(',');
```

Comment:
That is the nature of API at the boundary - it needs to validate its input. This is how all APIs in BCL behave - they validate their input even if another API that did the same validation was just called. From the performance point of the view the impact should be negligible because those pointers should be in CPU cache or even registers or even better optimizer can elide the second check altogether if it sees that the pointer is not changed by the code between the checks and it's not `volatile`. But if  you have a suggestion for improvement please share.

Reply:
> But if you have a suggestion for improvement please share.

Sure!

I would simply not use the extension method here - just keep it conventional. `kv?.Value == null` a couple of lines earlier already nullchecks, so I'd just do 

```
var pairs = kv.Value.Length == 0 ? 
```

The other advantage would be that both `kv?.Value == null` and `kv.Value.Length` are well known constructs for everyone, but the `IsEmpty()` extension method is something that only exists in this repo - that's why I looked into it it and started wondering like "that's not how I usually check on strings 🤔 ".

To your point:
> That is the nature of API at the boundary 

I would not create an API boundary in order to check if a string is null or empty.